app/models/language.rb

```diff
@@ -51,11 +51,26 @@ def explode_path_to_hash key, path
     {key => path}.merge Hash[(0...elements.size).map{|i|"#{key}[#{i}]".to_sym}.zip(elements)]
   end
 
+  # for validating submissions
   def self.submission_options
-    latest = LanguageGroup.where(identifier: %w[c++ c python haskell java ruby j]).pluck(:current_language_id)
-    old = Language.where(identifier: %w[c++11 c++14 c99 python2]).pluck(:id)
-    languages = Language.where(:id => latest).order(:identifier) + Language.where(:id => old).order(:identifier)
-    Hash[languages.map{ |language| ["#{language.name}", language.id] }]
+    grouped_submission_options.values.reduce(:merge)
+  end
+
+  # for the selection dropdown
+  def self.grouped_submission_options
+    current = ["c++", "c", "python", "java", "haskell", "ruby", "j"]
+    other = ["c++14", "c++11", "c99", "python2"]
```

Comment:
Maybe add comments to draw attention to the fact that one contains language group identifiers and the other contains language identifiers?